routes/api/events.js

```diff
@@ -1,9 +1,376 @@
+      dateEnd,
+      type,
+      location,
+      website,
+      image,
+      date,
+    } = req.body;
+
+    // Build event object
+    const eventFields = {};
+    eventFields.user = req.user.id;
+    if (title) eventFields.title = title;
+    if (description) eventFields.description = description;
+    if (dateStart) eventFields.dateStart = dateStart;
+    if (dateEnd) eventFields.dateEnd = dateEnd;
+    if (type) eventFields.type = type;
+    if (location) eventFields.location = location;
+    if (website) eventFields.website = website;
+    if (image) eventFields.image = image;
+    if (date) eventFields.date = date;
+
+    try {
+      let event = await Event.findOne({ user: req.user.id });
+
+      // Create
+      event = new Event(eventFields);
+
+      await event.save();
+      res.json(event);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send('Server error');
+    }
+  }
+);
+
+// @route   PUT api/events
+// @desc    Edit event
+// @access  Private
+router.put(
+  '/id/:id',
+  [
+    auth,
+    [
+      check('title', 'Title is required').notEmpty(),
+      check('description', 'Description is required').notEmpty(),
+      check('dateStart', 'Start date is required').notEmpty(),
+      check('dateEnd', 'End date is required').notEmpty(),
+      check('type', 'Type is required').notEmpty(),
+      check('location', 'Location is required').notEmpty(),
+    ],
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({
+        errors: errors.array(),
+      });
+    }
+
+    const {
+      title,
+      description,
+      dateStart,
+      dateEnd,
+      type,
+      location,
+      website,
+      image,
+      date,
+    } = req.body;
+
+    // Build event object
+    const eventFields = {};
+    eventFields.user = req.user.id;
+    if (title) eventFields.title = title;
+    if (description) eventFields.description = description;
+    if (dateStart) eventFields.dateStart = dateStart;
+    if (dateEnd) eventFields.dateEnd = dateEnd;
+    if (type) eventFields.type = type;
+    if (location) eventFields.location = location;
+    if (website) eventFields.website = website;
+    if (image) eventFields.image = image;
+    if (date) eventFields.date = date;
+
+    try {
+      const event = await Event.updateOne(
+        {
+          user: req.user.id,
+          _id: req.params.id,
+        },
+        {
+          $set: {
+            title,
+            description,
+            dateStart,
+            dateEnd,
+            type,
+            location,
+          },
+        }
+      );
+
+      res.json(event);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send('Server Error');
+    }
+  }
+);
+
+// @route   POST api/events/:id
+// @desc    Registration on event
+// @access  Public
+router.post(
+  '/id/:id',
+  [
+    check('name', 'Name is required').notEmpty(),

```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/spumony/xvent-project-backend/pullRequest?prid=7531905)
```suggestion
    check("name", 'Name is required').notEmpty(),
```